runtime/src/message_processor.rs

```diff
@@ -1037,27 +1054,23 @@ impl MessageProcessor {
 
         let pre_accounts = Self::create_pre_accounts(message, instruction, accounts);
         let program_id = instruction.program_id(&message.account_keys);
+        let keyed_accounts =
+            Self::create_keyed_accounts(message, instruction, executable_accounts, accounts);
         let mut invoke_context = ThisInvokeContext::new(
             program_id,
             rent_collector.rent,
             pre_accounts,
             executable_accounts,
             account_deps,
+            keyed_accounts.as_slice(),
```

Comment:
Could `as_slice()` be causing some of the lifetime challenges?
https://stackoverflow.com/questions/24542064/why-does-the-variable-not-live-long-enough

Reply:
It certainly does, and as I wrote in one of the other comments, in the end I want to move the actual object in here, not just the reference.